src/plugin.rs

```diff
@@ -108,21 +138,39 @@ impl MigrationSource<'static> for MigrationList {
 }
 
 #[command]
-async fn load(
+async fn load<R: Runtime>(
+    app: AppHandle<R>,
     db_instances: State<'_, DbInstances>,
     migrations: State<'_, Migrations>,
     db: String,
-) -> Result<()> {
-    if !Db::database_exists(&db).await.unwrap_or(false) {
-        Db::create_database(&db).await?;
+) -> Result<String> {
+    // fully qualified DB connection string
+    let fqdb = match cfg!(feature = "sqlite") {
+        true => path_mapper(app_path(&app), db.clone()),
+        false => db.clone(),
+    };
+
+    #[cfg(feature = "sqlite")]
+    create_dir_all(app_path(&app)).expect("Problem creating App directory!");
+
+    let db_exists = Db::database_exists(&fqdb).await.unwrap_or(false);
+
+    if !db_exists {
+        Db::create_database(&fqdb).await?;
```

Comment:
do you mean with the `let if` clause? I considered that because it's a bit more compact but it'll need to change more toward this syntax if we add in the ability for a user to configure whether the _migrations_ happen.